Fix canadarm's harcoded paths / update control interface / use xacro file

ana-GT opened the pull request:
Hi :). While trying to use the Canadarm, I ran into a few issues that were solved with the changes in this PR (plus this other [accompanying PR in the simulation package](https://github.com/space-ros/simulation/pull/9/files)). Changes are the following:

1. Use the canadarm's xacro file in the launch file
2. Change the type of controller from effort to position (JointGroupPositionController does exist in the position_controllers package).
3. Update simple.world to use model paths instead of hardcoded ones.

There is a video attached in the simulation's PR that shows the arm moving after using these changes. Tested in a ROS Humble environment.

quarkytale commented:
@Bckempa  I can take a look at this one and testing demos in general.

quarkytale commented on canadarm/worlds/simple.world:
```diff
@@ -47,7 +47,7 @@
             <geometry>
               <mesh>
-                <uri>/home/spaceros-user/demos_ws/install/simulation/share/simulation/models/canadarm/meshes/earth.dae</uri>
+                <uri>model://canadarm/meshes/earth.dae</uri>
```
As far as I recall, these were hard-coded to get the moveit_config working which has to be refactored anyway so I won't block on this.

quarkytale commented:
I'm still not able to get past the rqt issue to test the associated PR, and both of these should go in together.

ana-GT commented:
> I'm still not able to get past the rqt issue to test the associated PR, and both of these should go in together.

Apologies about not replying timely! On it now, will reply in the associated PR.

ana-GT commented:
Just an update on this: To make testing easier, I added a couple of additional changes since last time I sent this PR:

1. Added a joint_trajectory_controller in the control yaml, in addition to the already existing position controller.
2. Modified the launch file to load the joint_trajectory_controller
3. Added a **move_arm** node, similar to the one in the rover, so the user can send service calls for open_arm, close_arm and random_arm (a random pose). This should make it easier to test the changes.
4. Added a bit of documentation on regards to (3) in the README file.

Arm seems to be moving better (less bouncy) with the joint_trajectory_controller. I can revert this change back to using the position controller by default, in case there is a preference/reason to choose that one.  Quick video of the arm moving around is here:

https://github.com/space-ros/demos/assets/555379/486d0865-af10-46b3-9b22-40fa09e5df96

quarkytale commented:
@Bckempa this is ready to be merged!
